src/containers/Game/Game.js

```diff
@@ -0,0 +1,169 @@
+    };
+
+    componentDidMount() {
+        const { wordsNumber } = this.state;
+
+        const index = Math.floor(Math.random() * wordsNumber);
+        const word = words[index];
+        const lettersObjects = word.value.split('')
+            .map((letter) => ({
+                value: letter,
+                id: generateID(),
+                active: true,
+            }));
+
+        this.setState({
+            word,
+            wordLetters: lettersObjects,
+            shuffledLetters: this.shuffleArray(lettersObjects),
+        });
+    }
+
+    shuffleArray = (arr) => {
+        let currIndex = arr.length;
+        let tempValue;
+        let randomIndex;
+
+        while (currIndex) {
+            randomIndex = Math.floor(Math.random() * currIndex);
+            currIndex -= 1;
+
+            tempValue = arr[currIndex];
+            // eslint-disable-next-line
+            arr[currIndex] = arr[randomIndex];
+            // eslint-disable-next-line
+            arr[randomIndex] = tempValue;
+        }
+
+        return arr;
+    };
+
+    toggleLettersActiveState = (id) => {
+        const { wordLetters } = this.state;
+
+        const index = wordLetters.findIndex((el) => el.id === id);
+        wordLetters[index].active = !wordLetters[index].active;
+        this.setState({
+            wordLetters,
+        });
+    };
+
+    handleLetterClick = (id) => {
+        const { wordLetters, lettersInSlots } = this.state;
+        const letterSelected = wordLetters
+            .find((el) => el.id === id);
+
+        if (!letterSelected.active) {
+            return;
+        }
+
+        this.toggleLettersActiveState(id);
+
+        this.setState({
+            lettersInSlots: [...lettersInSlots, letterSelected],
+        });
+
+        if ([...lettersInSlots, letterSelected].length === 7) {
+            this.gameOver();
+        }
+    };
+
+    handleSlotClick = (id, i) => {
+        const { lettersInSlots } = this.state;
+
+        if (i !== lettersInSlots.length - 1) {
+            return;
+        }
+
+        this.toggleLettersActiveState(id);
+        lettersInSlots.pop();
+
+        this.setState({
+            lettersInSlots,
+        });
+    };
+
+    joinLetters = (arr) => arr.reduce((a, b) => a + b.value, '');
+
+    gameOver = () => {
+        this.setState({
+            // eslint-disable-next-line
+            isGameOn: false,
+        });
+
+        setTimeout(() => {
+            this.checkResult();
+        }, 500);
+    };
+
+    checkResult = () => {
+        const { word, lettersInSlots } = this.state;
+        const result = this.joinLetters(lettersInSlots);
+
+        if (result === word.value) {
+            this.setState({
+                // eslint-disable-next-line
+                isGameResultPositive: true,
+            });
+        } else {
+            this.setState({
+                // eslint-disable-next-line
+                isGameResultPositive: this.checkAnagrams(),
+            });
+        }
```

Comment:
lines 131-141: this is overly complicated, checking the value, then checking anagrams, see my comment regarding data structure

Reply:
👍